File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int secondNum = target - nums[i];
            System.out.println(map);
            if (map.containsKey(secondNum)) {
                return new int[]{map.get(secondNum), i};
            }
            map.put(nums[i], i);
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        while True:
            res = numbers[r] + numbers[l]
            if res < target:
                l += 1
            elif res > target:
                r -= 1
            else:
                break
        return [l+1, r+1]